### PR TITLE
Drop messages more than one year old

### PR DESCRIPTION
**Pull request by candeira**

### CODE REVIEW

File: plugins/to.py [hunk 1]
```diff
@@ -15,7 +15,15 @@ def setup(irc):
                 ("unixtime", int),
                 ("channel", str),
                 ("message", str))
-        # TODO: Drop messages more than a year old
+        else:
+            # Drop messages more than a year old
+            now = int(time.time())
+            one_year = 60 * 60 * 24 * 366
+            query = "SELECT * FROM saxo_to"
+            for row in db.query(query):
```
Comment (dpk) [inline plugins/to.py]: Wouldn't it be better to add a `WHERE` clause here instead of selecting all messages? Or even better just `DELETE .. WHERE`

**Comment by sbp** [comment]
I'd prefer that this:

1. Exports all old messages to a file
2. Is manually invoked using an administrator command

On the first point, ideally messages put into saxo should never be lost. This is a good guideline for computing in general. User data should never just go mysteriously missing.

On the second point, saxo functionality should as much as possible be through commands rather than plugins. Whether the garbage collection here should be done manually or automatically is down to preference, but the easiest way to support those who want to retain all messages is to make it manual.

File: plugins/to.py (hunk 1 above)
Comment (candeira) [inline plugins/to.py]: Yeah, you're right.

**Comment by candeira** [comment]
I have no horse in this race. I just found what I thought was an easy TODO to implement as I was looking at the code to put in the feature I really wanted.

**Pull request closed without merging**